soc/arm/nordic_nrf/Kconfig.defconfig

```diff
@@ -9,6 +9,15 @@ if SOC_FAMILY_NRF
 
 source "soc/arm/nordic_nrf/*/Kconfig.defconfig.series"
 
+if SYS_CLOCK_EXISTS
+# If the kernel has timer support, enable both clock control and timer
+config CLOCK_CONTROL
```

Comment:
Clock control might be used when timer is not used.
IMHO we cannot bind this feature to the kernel timer support.

Reply:
I can fix this, but it forces me to use a `select` statement in Kconfig which is always dangerous. I think this might be OK in this case though, so I will fix it.

Reply:
Actually on second thought, this is not a problem at all. Here we enable both clock control and nrf timer if the system clock is enabled in the kerne. If the timer support is not required then none will be enabled, and in that particular configuration the user can enable `CLOCK_CONTROL=y` himself.